src/main/java/com/wolox/training/services/BookService.java

```diff
@@ -0,0 +1,69 @@
+package com.wolox.training.services;
+
+import com.wolox.training.dtos.BookDTO;
+import com.wolox.training.exceptions.ObjectNotFoundException;
+import com.wolox.training.models.Book;
+import com.wolox.training.repositories.BookRepository;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly = true)
+public class BookService {
+
+    private static final String BOOK = "book";
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    public List<Book> findAll() {
+        return bookRepository.findAll();
+    }
+
+    public Optional<Book> findByAuthor(String bookAuthor) {
+        return bookRepository.findByAuthor(bookAuthor);
```

Comment:
You could add the .orElseThrow here :)
It will clear up the logic you added to the controller.